### PR TITLE
feat: improve influxdb v2 api compability

### PR DESCRIPTION
**Pull request by Fengys123**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This pr mainly improve influxdb v2 api compability.

- Write using the curl command as shown below
    ```shell
    curl -i -XPOST "http://localhost:4000/v1/influxdb/api/v2/write?bucket=public&precision=ms" \
    --data-binary \
    'monitor,host=127.0.0.1 cpu=0.1,memory=0.4 1667446797450
     monitor,host=127.0.0.2 cpu=0.2,memory=0.3 1667446798450
     monitor,host=127.0.0.1 cpu=0.5,memory=0.2 1667446798450'
    ```
- Write using Go sdk, an example [here](https://github.com/Fengys123/influxdb_write_example/blob/main/main.go)

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
fix https://github.com/GreptimeTeam/greptimedb/issues/1815

### CODE REVIEW

File: src/servers/src/http/influxdb.rs [hunk 1]
```diff
@@ -46,30 +46,40 @@ pub async fn influxdb_write(
     Query(mut params): Query<HashMap<String, String>>,
     lines: String,
 ) -> Result<impl IntoResponse> {
-    let db = params
```
Comment (killme2008) [inline src/servers/src/http/influxdb.rs:49]: I think we can refactor this function to accept the `db` parameter and add two new functions `influxdb_write_v1` and `influxdb_write_v2` that extract the `db` value from different params. 
Under high writing load, this logic to extract `db` value is an unnecessary cost.

**Review by killme2008 — APPROVED** [review]
Almost LGTM

File: src/servers/src/http.rs [hunk 2]
```diff
@@ -599,6 +599,7 @@ impl HttpServer {
     fn route_influxdb<S>(&self, influxdb_handler: InfluxdbLineProtocolHandlerRef) -> Router<S> {
         Router::new()
             .route("/write", routing::post(influxdb_write))
+            .route("/api/v2/write", routing::post(influxdb_write))
```
Comment (MichaelScofield) [inline src/servers/src/http.rs:602]: We have `/write` and `/api/v2/write`, they seem not consistent with each other. Is it able to change one? Like, do we have to use `/api/v2/write` to adapt influxdb's official sdk?

Comment (Fengys123) [reply src/servers/src/http.rs:602]: Influxdb go sdk v2 will add "/api/v2" after the url.
https://github.com/influxdata/influxdb-client-go/blob/114e28a215dfe0d9e0763e822c9b2e961b1f35fa/api/http/service.go#L62C1-L79C2

So we need two APIs. `/write` and `/api/v2/write`.

And when influxdb sdk builds the `client`, the required `addr` parameter can also be consistent.
https://github.com/Fengys123/influxdb_write_example/blob/main/main.go#L21
https://github.com/Fengys123/influxdb_write_example/blob/main/main.go#L52

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**